### ISSUE TITLE
[EPIC: Task/Feature/Issue] BharatGen Slakshana M0 PoC

### ISSUE THREAD

**Issue by deanwampler** [The-AI-Alliance/tapestry#189]
This epic covers the PoC lead by BharatGen using their Slackshana stack, for M0.

**Comment by deanwampler** [comment]
@bapichatterjee feel free to change this issue and sub-issues as you see fit, including assignments.

**Comment by kb-bhatta** [comment]
@msingh172 @bapichatterjee - please share the google doc here being developed on m0 to help us keep on same page

**Comment by kb-bhatta** [comment]
@bapichatterjee @msingh172 - let us know on review of the open issue to help fill in details

**Comment by bapichatterjee** [comment]
Hi @kb-bhatta: we have shared our proposal to the Monash team; hope to converge on the model and the task this week. Tomorrow, there will be another meeting to discuss the task. Will keep posted.

**Comment by bapichatterjee** [comment]
Here is the BharatGen-Monash (almost sure) training detail for M0 to start soon:

- We IFT Olmo (2 or 3) 7B model on the CultureInstruct dataset https://drive.google.com/file/d/139oNuyEVdvprEIWUBuBd4BcBrWMwp0Hw/view?usp=sharing that contains the dataset from multiple countries.
- We have split the dataset into 3 parts (data folder):
  - South Asian countries (including Afghanistan, Bangladesh, Bhutan, India, Maldives, Nepal, Pakistan, and Sri Lanka): 15K samples, 4.2M tokens (according to Olmo tokenizer).
  - Out-of-South-Asia countries: 294K samples, 75M tokens (according to Olmo tokenizer). We can split this part further by region. We estimate the Oceania region (Australia & NZ) has around 13K samples, 3.5M tokens.
  - Instruct data without country label: culture-relevant instructions without country label: 158K samples, 30M tokens (according to Olmo tokenizer).
  - Each datapoint contains the system prompt, user instruction, and assistant answers.
- The training happens using the Slakshna stack https://github.com/dcll-iiitd/Slakshna. We already tested the stack's connectivity across India and Australia by fine-tuning the LLama 1.1B model for several epochs.
- We evaluate the models on two sides on cultural benchmarks for which (for Olmo 2 7B) we have the numbers available in a centralized setting: https://aclanthology.org/2025.naacl-long.465.pdf
- For the Federated Training, we aim to determine the optimal hyperparameters for the following:
  1. The number of local steps before synchronization.
  2. The sparsity and quantization levels.
  3. The LR schedule for the optimizer.
  4. The NS constants for the Muon optimizer.
  5. The rank of the LoRA adaptor.
  6. Any other relevant HP.
To summarize, we aim to conduct cultural IFT in a geo-localized consortium training setting of a reasonably sized LLM.

**Comment by deanwampler** [comment]
The epic is finished!!
